packages/connector-naver/src/index.ts

```diff
@@ -0,0 +1,153 @@
+  GetAuthorizationUri,
+  GetConnectorConfig,
+  GetUserInfo,
+  SocialConnector,
+  validateConfig,
+} from '@logto/connector-core';
+import { assert, conditional } from '@silverhand/essentials';
+import got, { HTTPError } from 'got';
+
+import {
+  accessTokenEndpoint,
+  authorizationEndpoint,
+  defaultMetadata,
+  defaultTimeout,
+  userInfoEndpoint,
+} from './constant';
+import {
+  accessTokenResponseGuard,
+  authResponseGuard,
+  naverConfig,
+  naverConfigGuard,
+  userInfoResponseGuard,
+} from './types';
+
+const getAuthorizationUri =
+  (getConfig: GetConnectorConfig): GetAuthorizationUri =>
+  async ({ state, redirectUri }) => {
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig<naverConfig>(config, naverConfigGuard);
+
+    const queryParameters = new URLSearchParams({
+      client_id: config.clientId,
+      redirect_uri: redirectUri,
+      response_type: 'code',
+      state,
+    });
+
+    return `${authorizationEndpoint}?${queryParameters.toString()}`;
+  };
+
+export const getAccessToken = async (
+  config: naverConfig,
+  codeObject: { code: string; redirectUri: string }
+) => {
+  const { code, redirectUri } = codeObject;
+  const { clientId, clientSecret } = config;
+
+  // Note：Need to decodeURIComponent on code
+  // https://stackoverflow.com/questions/51058256/google-api-node-js-invalid-grant-malformed-auth-code
+  const httpResponse = await got.post(accessTokenEndpoint, {
+    form: {
+      code: decodeURIComponent(code),
+      client_id: clientId,
+      client_secret: clientSecret,
+      redirect_uri: redirectUri,
+      grant_type: 'authorization_code',
+    },
+    timeout: defaultTimeout,
+  });
+
+  const result = accessTokenResponseGuard.safeParse(JSON.parse(httpResponse.body));
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error.message);
+  }
+
+  const { access_token: accessToken } = result.data;
+
+  assert(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+  return { accessToken };
+};
+
+const getUserInfo =
+  (getConfig: GetConnectorConfig): GetUserInfo =>
+  async (data) => {
+    const { code, redirectUri } = await authorizationCallbackHandler(data);
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig<naverConfig>(config, naverConfigGuard);
+    const { accessToken } = await getAccessToken(config, { code, redirectUri });
+
+    try {
+      const httpResponse = await got.post(userInfoEndpoint, {
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        timeout: defaultTimeout,
+      });
+
+      const result = userInfoResponseGuard.safeParse(JSON.parse(httpResponse.body));
+
+      if (!result.success) {
+        throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error.message);
+      }
+
+      const { response } = result.data;
+      const { id, email, nickname, profile_image } = response;
```

Comment:
can we also get the phone number from the profile?
a follow-up question is: does the phone number come with a country code?

Reply:
If a mobile phone number is provided, Logto can choose to enable the user with passwordless sign-in via the phone number got from Naver profile.

Reply:
Yes, a developer can set phone number as necessary data from the user.

<img width="480" alt="image" src="https://user-images.githubusercontent.com/10739332/190369077-eb5a9824-4ac8-4258-b7d6-e8ae538f11cd.png">

However, the phone number does not come with a country code.

<img width="409" alt="image" src="https://user-images.githubusercontent.com/10739332/190369443-d4c9a97b-2cf1-4add-b34e-16caa6a3c0eb.png">

So, I think it is not good for passwordless sign-in.

Reply:
Cool! I can not receive a verification code message with my Chinese phone number either.
So let's keep it as it is. If only Korean numbers receive a verification code, we may add the Korean country code before phone numbers in Naver profiles by default in the future.